server/annotation/collection.ts

```diff
@@ -0,0 +1,105 @@
+import type { HydratedDocument, Types } from "mongoose";
+import type { Annotation } from "./model";
+import AnnotationModel from "./model";
+
+class AnnotationCollection {
+  /**
+   * Add a Annotation to the collection
+   *
+   * @param {string} authorId - The id of the author
+   * @param {string} lectureId - The id of the lecture the annotation corresponds to
+   * @param {Number} hour - The hour of the point in video the annotation correponds to
+   * @param {Number} minute - The minute of the point in video the annotation correponds to
+   * @param {Number} second - The second of the point in video the annotation correponds to
+   * @param {string} content - The content of the annotation
+   * @returns {Promise<HydratedDocument<Annotation>>} - The created Annotation
+   */
+  static async addOne(
+    authorId: Types.ObjectId | string,
+    lectureId: Types.ObjectId | string,
+    hour: Number,
+    minute: Number,
+    second: Number,
+    content: string
+  ): Promise<HydratedDocument<Annotation>> {
+    const date = new Date();
+    const annotation = new AnnotationModel({
+      authorId: authorId,
+      lectureId: lectureId,
+      hour: hour,
+      minute: minute,
+      second: second,
+      content: content,
+      dateCreated: date,
+    });
+
+    await annotation.save();
+    return annotation.populate("authorId");
+  }
+
+  /**
+   *
+   * Find all annotations with a given parent lecture with lectureId and authorId.
+   *
+   * @param {string} lectureId - The ID of the lecture
+   * @param {string} authorId - The ID of the annotation author
+
+   * @returns {Promise<Array<HydratedDocument<Annotation>>>} - An array of matching annotations
+   */
+  static async findAllByLectureAndAuthor(
+    lectureId: Types.ObjectId | string,
+    authorId: Types.ObjectId | string
+  ): Promise<Array<HydratedDocument<Annotation>>> {
+    return AnnotationModel.find({ lectureId: lectureId, authorId: authorId })
+      .sort("-date")
+      .populate("authorId");
+  }
+
+  /**
+   *
+   * Updates a annotation in the collection
+   *
+   * @param {string} annotationId - The ID of the annotation to be updated
+   * @param {string} content - The new content of the annotation
+   * @returns {Promise<HydratedDocument<Annotation>>} - The updated annotation
+   */
+  static async updateOne(
```

Comment:
How exactly is this going to work? It seems like you can update an existing annotation with a specific time to have a different time — how would this work on the front end? Intuitively I thought that once you made an annotation it was tied to that time, and any updates would update the annotation itself